sources/mongodb/helpers.py

```diff
@@ -45,32 +45,53 @@ def __init__(
             self.last_value = None
 
     @property
-    def _filter_op(self) -> Dict[str, Any]:
+    def _sort_op(self) -> List[Optional[Tuple[str, int]]]:
         if not self.incremental or not self.last_value:
+            return []
+        if self.incremental.last_value_func is max:
+            return [(self.cursor_field, ASCENDING)]
+        elif self.incremental.last_value_func is min:
+            return [(self.cursor_field, DESCENDING)]
+        return []
```

Comment:
The `_sort_op()` method moved without changes into the parent class to make sorting work in both parallel and ordinary cases

Reply:
take both row order and last value func. look here for code and tests (sql database didn't do that correctly as well): https://github.com/dlt-hub/verified-sources/pull/476

this is due to python code that applies both `row_order` and `last_value_func` to find out if items are out of range